docs/source/Diff.rst

```diff
@@ -0,0 +1,74 @@
+.. _job_diff_:
+
+========
+Job Diff
+========
+
+Avocado Diff plugin allows users to easily compare several aspects of
+two given jobs.
+
+The basic usage is::
+
+    $ avocado diff 7025aaba 384b949c
+    --- 7025aaba9c2ab8b4bba2e33b64db3824810bb5df
+    +++ 384b949c991b8ab324ce67c9d9ba761fd07672ff
+    @@ -1,15 +1,15 @@
+     
+     COMMAND LINE
+    -/usr/bin/avocado run sleeptest.py
+    +/usr/bin/avocado run passtest.py
+     
+     TOTAL TIME
+    -1.00251293182
+    +0.000292062759399
+     
+     VARIANTS
+     Variant 1: /
+     
+     TEST RESULTS
+    -1-sleeptest.py:SleepTest.test: PASS
+    +1-passtest.py:PassTest.test: PASS
+     
+     AVOCADO SETTINGS
+     [datadir.paths]
+
+Avocado Diff can compare and create an unified diff of:
+
+- Command line.
+- Job time.
+- Variants and parameters.
+- Tests results.
+- Configuration.
+- Sysinfo pre and post.
+
+You can enable/disable those items with ``--diff-filter``. Avocado Diff
+uses ``all,-sysinfo`` as defaults to ``--diff-filter``. Please see
+``avocado diff --help`` for more information.
+
+Besides the unified diff, you can also generate an html diff file and,
+optionally, open it on your preferred browser::
+
+    $ avocado diff 7025aaba 384b949c --html --open-browser
+    --- 7025aaba9c2ab8b4bba2e33b64db3824810bb5df
+    +++ 384b949c991b8ab324ce67c9d9ba761fd07672ff
+    @@ -1,15 +1,15 @@
+     
+     COMMAND LINE
+    -/usr/bin/avocado run sleeptest.py
+    +/usr/bin/avocado run passtest.py
+     
+     TOTAL TIME
+    -1.00251293182
+    +0.000292062759399
+     
+     VARIANTS
+     Variant 1: /
```

Comment:
Simply filter-out the sections which are the same, just as if one was manually using --diff-filter. The result will be like this:

--- aa416ece963381fb427431002aabea7aa99e3f4f
+++ 166a527461944518899352f1cfd88b36bb6d45f1
@@ -1,13 +1,12 @@
 
 COMMAND LINE
-./scripts/avocado run /bin/true /bin/false /bin/true /bin/false passtest.py
+./scripts/avocado run passtest.py failtest.py /bin/true /bin/false
 
 TOTAL TIME
-0.03422498703
+0.0232672691345
 
 TEST RESULTS
-1-/bin/true: PASS
-2-/bin/false: FAIL
+1-passtest.py:PassTest.test: PASS
+2-failtest.py:FailTest.test: FAIL
 3-/bin/true: PASS
 4-/bin/false: FAIL
-5-passtest.py:PassTest.test: PASS